### PR TITLE
CO-1928 Tests for sms_sponsorship

### PR DESCRIPTION
**Pull request by quentingigon**

### CODE REVIEW

**Comment by quentingigon** [comment]
Travis test was failing because US servers were down

File: sms_sponsorship/tests/test_sms_compassion.py [hunk 1]
```diff
@@ -19,12 +19,112 @@ class TestSmsCompassion(BaseSponsorshipTest):
 
     def setUp(self):
         super(TestSmsCompassion, self).setUp()
-        self.child = self.create_child('UG1119182')
-        self.child.hold_id.channel = 'sms'
+        self.child = self.env['compassion.child'].search([
+            ('id', '=', 5)
```
Comment (ecino) [inline sms_sponsorship/tests/test_sms_compassion.py]: Never browse by id in test, you never know what the id will be on another test database... Why did you change how it was done previously?

File: sms_sponsorship/tests/test_sms_compassion.py [hunk 2]
```diff
@@ -19,12 +19,112 @@ class TestSmsCompassion(BaseSponsorshipTest):
...
+        ])
+        # self.child.hold_id.channel = 'sms'
+        self.partner = self.env['res.partner'].search([
+            ('id', '=', 15)
```
Comment (ecino) [inline sms_sponsorship/tests/test_sms_compassion.py]: Not a good practice, you should use self.env.ref() instead and find a xml_id in the demo data of base module.

File: sms_sponsorship/tests/test_sms_compassion.py [hunk 3]
```diff
@@ -19,12 +19,112 @@ class TestSmsCompassion(BaseSponsorshipTest):
...
+        ])
+        self.event = self.env['crm.event.compassion'].search([
+            ('id', '=', 1030)
```
Comment (ecino) [inline sms_sponsorship/tests/test_sms_compassion.py]: This will clearly fail on empty databases.

**Review by ecino — CHANGES_REQUESTED** [review]

File: sms_sponsorship/tests/test_sms_compassion.py (hunk 1 above)
Comment (quentingigon) [reply sms_sponsorship/tests/test_sms_compassion.py]: It didn't work with create_child method 
(Error: IntegrityError: duplicate key value violates unique constraint "compassion_project_icp_id_uniq")

**Comment by ecino** [comment]
Oops, again confilcts :( tests are failing as well.

**Comment by quentingigon** [comment]
My tests are working fine locally and fail on the server. I have this error : 
UserError: ('Please give connect_url and connect_api_key values in your Odoo configuration file.', ''). But there are values for those fields in odoo.conf..

**Comment by ecino** [comment]
Odoo.conf is not committed and not available to travis (this would reveal all our connection endpoints with username password, so please don't commit it). You should therefore have tests that never use them or mock them to avoid using real connections.

**Review by ecino — APPROVED** [review]
